exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -74,23 +83,35 @@ func NewPrwExporter(namespace string, endpoint string, client *http.Client, exte
 	}
 
 	userAgentHeader := fmt.Sprintf("%s/%s", strings.ReplaceAll(strings.ToLower(buildInfo.Description), " ", "-"), buildInfo.Version)
-
-	return &PrwExporter{
+	prwe := &PrwExporter{
 		namespace:       namespace,
 		externalLabels:  sanitizedLabels,
 		endpointURL:     endpointURL,
 		client:          client,
 		wg:              new(sync.WaitGroup),
 		closeChan:       make(chan struct{}),
+		walConfig:       walCfg,
 		userAgentHeader: userAgentHeader,
-	}, nil
+	}
+
+	if err := prwe.turnOnWALIfEnabled(); err != nil {
+		return nil, err
+	}
```

Comment:
This should happen in `Start` of the component not during creation.

Reply:
AFAIK there isn't a .Start for this exporter.

Reply:
Then you should add one :)